src/wallet_sync.rs

```diff
@@ -166,15 +167,15 @@ impl SwapCoin {
 }
 
 impl Wallet {
-    pub fn save_new_wallet_file(
-        wallet_file_name: &str,
+    pub fn save_new_wallet_file<P: AsRef<Path>>(
+        wallet_file_name: P,
```

Comment:
Why do you this generic parameter? Why not write it as `wallet_file_name: AsRef<Path>`. Genuinely asking as I'm new to rust

Reply:
I'm thinking about this as one of those "idiomatic" [patterns](https://rust-lang.github.io/api-guidelines/flexibility.html?highlight=AsRef#functions-minimize-assumptions-about-parameters-by-using-generics-c-generic) that tend to proliferate through codebases. Here we want to express that the function can accept anything which could be a reference to the `Path`. Usually it's `PathBuf`, `String`, `OsStr` IIRC...

It's impossible (since edition 2018, IIRC) to use bare traits in type position like you wrote: `file: AsRef<Path>`. If you want to express something similar, some options could be is to instead use `wallet_file_name: impl AsRef<Path>` or `wallet_file_name: dyn AsRef<Path>`. Both options are more verbose than generic, IMO, and less common in the existing code (like stdlib). Moreover, the second one uses dynamic dispatch which is kind of less efficient and frowned upon for this use-case.